File: js/plain/p2.js
'use strict'

var f = 1
var fLast = 1

var sum = 0

while (f < 4000000)
{
    var tmp = f
    f += fLast
    fLast = tmp

    if (f % 2 == 0)
    {
        sum += f
    }
}

console.log(sum)

File: go/p2.go
package main

import "fmt"

func main() {

	f := 2
	f_last := 1

	sum := 0

	for {
		if f%2 == 0 {
			sum += f
		}

		f, f_last = f+f_last, f

		if f > 4000000 {
			break
		}
	}

	fmt.Println(sum)
}
